### PR TITLE
edi SketcherFixAllPoints macro and icon

### PR DESCRIPTION
**Pull request by edi271**
Thank you for creating a pull request to contribute to FreeCAD-macros!
To integrate your macro please make sure the following steps are complete:

- [ ] Please check this box if you're not submitting a new macro.   
- [ ] Are you submitting a new macro ?  
- [ ] Have you followed the ['How to submit a macro'](../README.md#how-to-submit-a-macro) section of the README.md ?  
- [ ] Your macro has a [Description](../README.md#macro-description) in its header.  
- [ ] Your macro has a [CamelCase name](../README.md#camelcase-macro-name).  
- [ ] Your macro is named [appropriately](../README.md#macro-name-specifics).  
- [ ] Your macro contains a [Metadata section](../README.md#macro-metadata) that immediately follows the header description.  
- [ ] Your macro is Python3/Qt5 compliant and tested on the latest FreeCAD stable and development releases.  
- [ ] You're including documentation on how your macro works (bonus: screenshots and/or video on the Wiki)  
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)  
- [ ] Commit message is titled in the following way `[MacroName] Short description`.
- [ ] Optional, write or update the changelog in the macro, from latest to oldest.

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

File: Sketcher/SketcherFixAllPoints.FCMacro [hunk 1]
```diff
@@ -0,0 +1,102 @@
+__Name__   = "SketcherFixAllPoints"
+__Author__  = "edi"
+__Version__ = "0.0.2"
+__License__ = 'License identifier from https://spdx.org/licenses/, e.g. LGPL-2.0-or-later as FreeCAD, MIT, CC0-1.0'
```
Comment (hyarion) [inline Sketcher/SketcherFixAllPoints.FCMacro:4]: No license has been selected

File: Sketcher/SketcherFixAllPoints.FCMacro [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	'''diameter constraints for all circles'''
+	geoList = Sketch.Geometry
+	for i in range(Sketch.GeometryCount):
+		if geoList[i].TypeId == 'Part::GeomCircle':
+			Sketch.addConstraint(Sketcher.Constraint('Diameter',i,geoList[i].Radius*2))
+
+AllPoints = []
+Sketch = getSketch()
+if Sketch != None:
+	deleteConstraints(Sketch)
```
Comment (hyarion) [inline Sketcher/SketcherFixAllPoints.FCMacro:98]: How about a showing a dialog if the user already have constraints in the sketch so nothing is lost without a warning? Maybe something like: "Running SketcherFixAllPoints will remove all your existing constraints. Do you want to proceed?"

**Review by hyarion — CHANGES_REQUESTED** [review]

File: Sketcher/SketcherFixAllPoints.FCMacro (hunk 1 above)
Comment (edi271) [reply Sketcher/SketcherFixAllPoints.FCMacro]: done

File: Sketcher/SketcherFixAllPoints.FCMacro (hunk 2 above)
Comment (edi271) [reply Sketcher/SketcherFixAllPoints.FCMacro:98]: I have extended the macro description. This macro is not suitable to manipulate handmade sketches having some missing constraints. It is intended to constrain automatic created, incomplete contrained sketches. The macro in use can be seen in https://www.youtube.com/watch?v=rDhbWcVTkMA (created by user thomas-neemann)

Comment (hyarion) [reply Sketcher/SketcherFixAllPoints.FCMacro:98]: It's much better after your change to the description

**Review by hyarion — APPROVED** [review]
✅ Tested on a simple sketch and it works on latest FreeCAD master branch
 ✅ The macro has a well written description
 ✅ The name is on the correct format and explains what it does (Maybe "FixateAllPoints" could have been slightly better but not really needed to change in my opinion)

File: Sketcher/SketcherFixAllPoints.FCMacro [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+__Author__  = "edi"
+__Version__ = "0.0.2"
+__License__ = 'LGPL v2'
+__Web__ = 'https://forum.freecadweb.org/viewtopic.php?f=13&t=52451'
+__Help__ = 'Open a sketch. Start the macro. All geometry is fixed by constraints.'
+__Status__ = 'Stable'
+__Requires__ = 'e.g. FreeCAD >= v0.19'
+__Date__    = "2021-01-07"
+"""
+This macro is intended to constrain automatic created, incomplete contrained sketches.
```
[the remark this answers is not in the document]
Comment (edi271) [reply Sketcher/SketcherFixAllPoints.FCMacro:11]: Its OK for me. (I am no English native speaker)

File: Sketcher/SketcherFixAllPoints.FCMacro [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+		if len(Pt.Ident) == 2:
+			Sketch.addConstraint(Sketcher.Constraint('Coincident',Pt.Ident[0][0],Pt.Ident[0][1],Pt.Ident[1][0],Pt.Ident[1][1]))
+
+def makeDiameters(Sketch):
+	'''diameter constraints for all circles'''
+	geoList = Sketch.Geometry
+	for i in range(Sketch.GeometryCount):
+		if geoList[i].TypeId == 'Part::GeomCircle':
+			Sketch.addConstraint(Sketcher.Constraint('Diameter',i,geoList[i].Radius*2))
+
```
Comment (galou) [inline Sketcher/SketcherFixAllPoints.FCMacro:94]: Please add the imports
```
import FreeCAD as app
import FreeCADGui as gui
```
and use `app` and `gui` in place of `App` and `Gui`.

File: Sketcher/SketcherFixAllPoints.FCMacro [hunk 5]
```diff
@@ -0,0 +1,102 @@
+__Name__   = "SketcherFixAllPoints"
+__Author__  = "edi"
+__Version__ = "0.0.2"
+__License__ = 'LGPL v2'
+__Web__ = 'https://forum.freecadweb.org/viewtopic.php?f=13&t=52451'
+__Help__ = 'Open a sketch. Start the macro. All geometry is fixed by constraints.'
+__Status__ = 'Stable'
+__Requires__ = 'e.g. FreeCAD >= v0.19'
+__Date__    = "2021-01-07"
```
Comment (galou) [inline Sketcher/SketcherFixAllPoints.FCMacro:9]: Please add `__Files__ = 'SketcherFixAllPoints.svg'`.

**Comment by galou** [comment]
Thanks again for your contribution!

Thanks @hyarion for the review!

**Pull request merged**